eth/api_backend.go

```diff
@@ -87,6 +109,32 @@ func (b *EthAPIBackend) BlockByNumber(ctx context.Context, blockNr rpc.BlockNumb
 	return b.eth.blockchain.GetBlockByNumber(uint64(blockNr)), nil
 }
 
+func (b *EthAPIBackend) BlockByNumberOrHash(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (*types.Block, error) {
+	if blockNr, ok := blockNrOrHash.Number(); ok {
```

Comment:
Good catch, currently the spec does not mention precedence or have any test cases for it.

 @charles-cooper care to update the spec regarding this scenario?  I see four options:

- Option 1: Block Number takes priority
- Option 2: Block Hash takes priority
- Option 3: If both blockNumber and blockHash are specified, return an error indicating these values are exclusive
- Option 4: If both blockNumber and blockHash are specified, enforce that both values are correct, otherwise return an error (i.e. lookup by hash, and return an error if returned header.Number != blockNumber).

Reply:
I opened https://github.com/ethereum/EIPs/issues/2247 to solicit feedback on which option people prefer.

Reply:
As Charles mentioned on the linked issue, probably makes sense to stay compatible with the behavior in EIP-234 (the one that added `blockHash` to eth_getLogs and friends), and that EIP says the number and hash arguments are mutually exclusive, so I'll update my code to Option 3.

Reply:
I've updated `BlockNumberOrHash.UnmarshalJSON` to return an error if both number and hash are specified with an error very similar to the one returned by `FilterCriteria.UnmarshalJSON` when both are specified as part of an EIP-234 request.